components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java

```diff
@@ -716,6 +716,45 @@ public void unregisterDeviceManagementService(DeviceManagementService deviceMana
         }
     }
 
+
+    public List<Device> getDevicesByStatus(EnrolmentInfo.Status status) throws DeviceManagementException {
+        List<Device> devices = new ArrayList<Device>();
+        List<Device> allDevices;
+        try {
+            DeviceManagementDAOFactory.getConnection();
+            int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+            allDevices = deviceDAO.getDevicesByStatus(status, tenantId);
+
+        } catch (DeviceManagementDAOException e) {
+            String errorMsg = "Error occurred while fetching the list of devices that matches to status: '"
+                              + status + "'";
+            log.error(errorMsg, e);
+            throw new DeviceManagementException(errorMsg, e);
+        } finally {
+
```

Comment:
Fixed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/00807e8295c06fc4241caebcc514319ae0d88f3b